File: Assignment11/C/Program3.c
// PS - Accept N number from user and accept one another number as NO, return index of last occurance of that number

#include<stdio.h>
#include<stdlib.h>

int CheckNum(int Arr[], int iLength, int start, int end) {
    int iCnt = 0;
    int index = 0;

    for(iCnt = 0; iCnt < iLength; iCnt++){
        if(Arr[iCnt] == No) {
            index = iCnt;
        }
    }

    return index;
}

int main() {
    int iSize = 0, start = 0, end = 0, iCnt = 0;
    int *ptr = NULL;
    int iRet = 0;

    printf("Enter number of elements that you want to store in array \n");
    scanf("%d", &iSize);

    printf("Enter start range \n");
    scanf("%d", &start);

    printf("Enter start range \n");
    scanf("%d", &end);

    ptr = (int *)malloc(sizeof(int) * iSize);

    printf("Enter elemets in array \n");
    for(iCnt = 0; iCnt < iSize; iCnt++) {
        scanf("%d", &ptr[iCnt]);
    }

    iRet = CheckNum(ptr, iSize, start, end);

    printf("1st occurance of %d is %d", iValue, iRet);

    free(ptr);

    return 0;
}

File: Assignment36/Java/Program3.java
// PS - write java program which accept N numbers from user and accept one more number and return index of last occurance of that number
import java.util.*;

class ArrayX
{
    public int CheckNumIndex(int Arr[], int No)
    {
        int last = 0;
        int occ = 0;
        for(int iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            if(Arr[iCnt] == No)
            {
                last = iCnt;
                occ++;
            }
        }
        if(occ == 0)
        {
            return -1;
        }
        else 
        {
            return last;
        }
    }
}

class Program3
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number of elements");
        int iSize = sobj.nextInt();

        int Arr[] = new int[iSize];

        System.out.println("Enter elements of array");
        for(int iCnt = 0; iCnt < iSize; iCnt++)
        {
            Arr[iCnt] = sobj.nextInt();
        }

        System.out.println("Enter number that you want to search");
        int No = sobj.nextInt();

        ArrayX aobj = new ArrayX();
        int iret = aobj.CheckNumIndex(Arr, No);
        if(iret == -1)
        {
            System.out.println("number is not present");
        } else
        {
            System.out.println("last occurance is at index :"+iret);
        }
    }
}